index.js

```diff
@@ -120,14 +120,24 @@ class Topic {
       return topics
     }
 
-    for (const item of list.children) {
+    const children = Array.from(list.children)
+
+    for (const [index, item] of children.entries()) {
       const label = item.querySelector('label')
       const checkbox = item.querySelector('input')
       if (label instanceof HTMLLabelElement && checkbox instanceof HTMLInputElement) {
         let childList = item.querySelector('ul')
         childList = childList instanceof HTMLUListElement ? childList : null
 
-        const topic = new Topic(label, checkbox, childList, parent)
+        const previous = index > 0 ? topics[index - 1] : null
+        const next = null
```

Comment:
This reads quite oddly - we define a `next` parameter to the `Topic` constructor but we never actually use it. We're always passing `null` and then manually manipulating the `next` property of the previously created `Topic` afterwards.

This might be better handled with a call to a separate function, so that line 145 becomes `return linkPreviousAndNext(topics)`.

Reply:
Alternatively you could carry on manipulating the `next` property of the previously created `Topic`, but without ever passing `next` in the first place (and removing `next` from the list of parameters), so that it is only ever defined after `Topic` has been created.

Reply:
Agree, I think I'll go with the second suggestion as it seems easier to follow